packages/contracts/contracts/ChugSplashManager.sol

```diff
@@ -839,44 +868,29 @@ contract ChugSplashManager is
         if (_amount == 0) {
             revert AmountMustBeGreaterThanZero();
         }
-        if (executorDebt[msg.sender] < _amount) {
+        if (executorDebt[_msgSender()] < _amount) {
             revert InsufficientExecutorDebt();
         }
+        if (_amount + totalProtocolDebt > address(this).balance) {
+            revert InsufficientFunds();
+        }
 
-        executorDebt[msg.sender] -= _amount;
+        executorDebt[_msgSender()] -= _amount;
         totalExecutorDebt -= _amount;
 
-        emit ExecutorPaymentClaimed(msg.sender, _amount, executorDebt[msg.sender]);
+        emit ExecutorPaymentClaimed(_msgSender(), _amount, executorDebt[_msgSender()]);
 
-        (bool success, ) = payable(msg.sender).call{ value: _amount }(new bytes(0));
-        if (!success) {
+        (bool paidExecutor, ) = payable(_msgSender()).call{ value: _amount }(new bytes(0));
+        if (!paidExecutor) {
             revert WithdrawalFailed();
         }
 
-        registry.announce("ExecutorPaymentClaimed");
-    }
-
-    /**
-     * @notice Allows the protocol creators to claim their royalty, which is only earned during
-       remotely executed deployments.
-     */
-    function claimProtocolPayment() external {
-        if (!managedService.hasRole(PROTOCOL_PAYMENT_RECIPIENT_ROLE, msg.sender)) {
```

Comment:
I updated the executor payment claiming logic to also send funds to the managed service contract and moved the withdraw logic over to that contract.